nifi-mock/src/main/java/org/apache/nifi/util/StandardProcessorTestRunner.java

```diff
@@ -229,33 +226,32 @@ public void run(final int iterations, final boolean stopOnFinish, final boolean
                 executorService.awaitTermination(runWait, TimeUnit.MILLISECONDS);
             } catch (final InterruptedException e1) {
             }
+        }
 
-            int finishedCount = 0;
-            boolean unscheduledRun = false;
-            for (final Future<Throwable> future : futures) {
-                try {
-                    final Throwable thrown = future.get(); // wait for the result
-                    if (thrown != null) {
-                        throw new AssertionError(thrown);
-                    }
+        int finishedCount = 0;
+        boolean unscheduledRun = false;
+        for (final Future<Throwable> future : futures) {
+            try {
+                final Throwable thrown = future.get(); // wait for the result
+                if (thrown != null) {
+                    throw new AssertionError(thrown);
+                }
 
-                    if (++finishedCount == 1 && stopOnFinish) {
-                        unscheduledRun = true;
-                        unSchedule();
-                    }
-                } catch (final Exception e) {
+                if (++finishedCount == 1 && stopOnFinish) {
+                    unscheduledRun = true;
+                    unSchedule();
                 }
+            } catch (final Exception e) {
+                e.printStackTrace();
```

Comment:
This use of `printStackTrace()` seems problematic, better throw a more particular type of exception?

Reply:
The `Exception` capture is nothing new, just previously it was being silently ignored, so I thought I'd at least indicate that the `catch` had been triggered, without potentially causing problems for lots of tests across the existing estate

Happy to revert this and silently swallow the `InterruptedException` & `ExecutionException`s, although I'll change it from catching all `Exception`s at the least

This approach matches other similar handling elsewhere within the `TestRunner`

Reply:
Thanks, I think it is best to avoid the `printStackTrace()` for now.